pkg/controller/acicontainersoperator.go

```diff
@@ -353,48 +408,57 @@ func (t *OperatorHandler) ObjectCreated(obj interface{}) {
 
 	log.Info("Platform flavor is ",acicontainersoperator.Spec.Flavor)
 
-	//if (acicontainersoperator.Spec.Flavor == "openshift-4.3") {
-	//
-	//	log.Info("Updating Network Status of the Openshift Object....")
-	//	cfg, err := config.GetConfig()
-	//	scheme := runtime.NewScheme()
-	//	err = networkv1.Install(scheme)
-	//	if err != nil {
-	//		log.Error(err)
-	//		return
-	//	}
-	//
-	//	rclient, err := client.New(cfg, client.Options{Scheme: scheme})
-	//	if err != nil {
-	//		return
-	//	}
-	//
-	//	clusterConfig := &configv1.Network{
-	//		TypeMeta:   metav1.TypeMeta{APIVersion: configv1.GroupVersion.String(), Kind: "Network"},
-	//		ObjectMeta: metav1.ObjectMeta{Name: "cluster"},
-	//	}
-	//
-	//	err = rclient.Get(context.TODO(), types.NamespacedName{
-	//		Name: "cluster",
-	//	}, clusterConfig)
-	//	if err != nil {
-	//		log.Info(err)
-	//	}
-	//	log.Info("Config Spec is  ", clusterConfig.Spec)
-	//	clusterConfig.Status.ClusterNetwork = clusterConfig.Spec.ClusterNetwork
-	//	clusterConfig.Status.NetworkType = clusterConfig.Spec.NetworkType
-	//	clusterConfig.Status.ServiceNetwork = clusterConfig.Spec.ServiceNetwork
-	//
-	//	log.Info("Status is ", clusterConfig.Status)
-	//
-	//	ctx := context.TODO()
-	//	err = rclient.Update(ctx, clusterConfig)
-	//	if err != nil {
-	//		log.Info(err)
-	//		return
-	//	}
-	//}
+	if (acicontainersoperator.Spec.Flavor >= "openshift-4.3") {
```

Comment:
What does ">" mean here (since you are comparing strings)? Instead it would be preferable if you can define an array (with currently only one string) which can be later extended to add future openshift versions as we support them. The definition of this should perhaps go into config.go.

Reply:
This is according to pervious suggested comment , and it is like kubernetes for which it fails and will be supported for openshift present/later versions like openshift-4.3/4.4  & so on. No need for an extra array to increase space complexity

Reply:
I am not very comfortable with the assumption that as far as this flavor value is concerned you are assuming it's either starts with "openshift" or "kubernetes". Moreover this assumption seems to be embedded deep in the code here.

You don't have to use an array, but "space" would hardly be the reason for not using it. ;-)